core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -887,6 +891,23 @@ class RDDSuite extends FunSuite with SharedSparkContext {
     assert(ancestors6.count(_.isInstanceOf[CyclicalDependencyRDD[_]]) === 3)
   }
 
+  test("parallelize with exception thrown on serialization should not hang") {
```

Comment:
this name is a little too specific. I'd leave out the parallelize and just call this something like
```
serialization exception should not hang scheduler
```